### PR TITLE
[2.0.14] handle SearchBox invalid template, issue #1187

### LINKED ISSUE #1187: SearchBox invalid template should be handled
This line should be prevented if scope.template is undefined.

https://github.com/angular-ui/angular-google-maps/blob/master/src/coffee/directives/search-box.coffee#L41

Otherwise an error at 
https://github.com/angular/angular.js/blob/77d8ae1d458ce526f77d059bae05100299d10f67/src/ng/urlUtils.js#L101
Is hit and protocol is null. This results when the url is undefined.
```
function urlIsSameOrigin(requestUrl) {
  var parsed = (isString(requestUrl)) ? urlResolve(requestUrl) : requestUrl;
  return (parsed.protocol === originUrl.protocol &&
          parsed.host === originUrl.host);
```
@kandizzy

### PR DESCRIPTION
**Pull request by kandizzy**
Checks that the search-box directive template attribute is defined. Resolves #1187

### CODE REVIEW

**Comment by cthrax** [comment]
LGTM.

File: src/coffee/directives/search-box.coffee [hunk 1]
```diff
@@ -38,6 +38,9 @@ angular.module('uiGmapgoogle-maps')
       
       link: (scope, element, attrs, mapCtrl) =>
         GoogleMapApi.then (maps) =>
+          unless scope.template?
```
Comment (nmccready) [inline src/coffee/directives/search-box.coffee]: this works , but would it not be better to watch scope.template until it is defined?

```
d = $q.defer()
promise = d.promise
unbind = scope.watch 'template', (newVal) ->
    d.resolve()

promise.then ->
   $http.get(scope.template, { cache: $templateCache })
             .success (template) =>
               if angular.isUndefined scope.events
```

Comment (kandizzy) [inline src/coffee/directives/search-box.coffee]: Another option is to set a default template that's just and HTML input field. What do you think?

Comment (nmccready) [inline src/coffee/directives/search-box.coffee]: Sure that would work for now.

**Comment by nmccready** [comment]
LGTM

**Pull request merged**